server/app/services/applications/ProgramAdminApplicationService.java

```diff
@@ -53,16 +84,90 @@ public Optional<Application> getApplication(long applicationId, ProgramDefinitio
    *
    * @param admin The Account that instigated the change.
    */
-  public void setStatus(Application application, StatusEvent newStatus, Account admin) {
+  public void setStatus(Application application, StatusEvent newStatusEvent, Account admin)
+      throws StatusNotFoundException {
+    Program program = application.getProgram();
+    String programName = program.getProgramDefinition().adminName();
+    Applicant applicant = application.getApplicant();
+    String newStatusText = newStatusEvent.statusText();
+    // The send/sent phrasing is a little weird as the service layer is converting between intent
+    // and reality.
+    boolean sendEmail = newStatusEvent.emailSent();
+
+    Optional<Status> statusDefMaybe =
+        program.getStatusDefinitions().getStatuses().stream()
+            .filter(s -> s.statusText().equals(newStatusText))
+            .findFirst();
+    if (statusDefMaybe.isEmpty()) {
+      throw new StatusNotFoundException(newStatusText, program.id);
+    }
+    Status statusDef = statusDefMaybe.get();
+
     ApplicationEventDetails details =
         ApplicationEventDetails.builder()
             .setEventType(ApplicationEventDetails.Type.STATUS_CHANGE)
-            .setStatusEvent(newStatus)
+            .setStatusEvent(newStatusEvent)
             .build();
     ApplicationEvent event = new ApplicationEvent(application, admin, details);
+
+    // Send email if requested and present.
+    if (sendEmail && statusDef.localizedEmailBodyText().isPresent()) {
+      // Notify an Admin/TI if they applied.
+      Optional<String> adminSubmitterEmail = application.getSubmitterEmail();
+      if (adminSubmitterEmail.isPresent()) {
+        sendAdminSubmitterEmail(programName, applicant, newStatusText, adminSubmitterEmail);
+      }
+      // Notify the applicant.
+      Optional<String> applicantEmail =
+          applicantService.getEmail(application.getApplicant().id).toCompletableFuture().join();
+      if (applicantEmail.isPresent()) {
+        sendApplicantEmail(programName, applicant, statusDef, applicantEmail);
+      }
+    }
+
     eventRepository.insertSync(event);
   }
 
+  private void sendApplicantEmail(
+      String programName, Applicant applicant, Status statusDef, Optional<String> applicantEmail) {
+    String civiformLink = baseUrl;
+    Locale locale = applicant.getApplicantData().preferredLocale();
+    String emailBody =
+        String.format(
+            "%s\n\nLog in to CiviForm at %s.",
+            statusDef.localizedEmailBodyText().get().getOrDefault(locale), civiformLink);
+    emailClient.send(
+        isStaging ? stagingApplicantNotificationMailingList : applicantEmail.get(),
+        String.format(STATUS_UPDATE_EMAIL_SUBJECT_FORMAT, programName),
```

Comment:
Blocking: The variable of `programName` looks to be correct here. However, it was initialized to `adminName`, which is the internal facing name of the program. I believe we should use `localizedName` for the current user's locale.

I think spotting these types of bugs might be clearer if the method took a `ProgramDefinition` as an argument and directly used `program.localizedName()...` since there's a single callsite. Passing a string feels a bit lossy to me (same commentary for the TI email below).